liquibase-cli/src/main/java/liquibase/integration/commandline/LiquibaseCommandLine.java

```diff
@@ -0,0 +1,789 @@
+        CommandLine.ParseResult parseResult = commandLine.getParseResult();
+
+        while (parseResult != null) {
+            if (parseResult.isUsageHelpRequested() || parseResult.isVersionHelpRequested()) {
+                return true;
+            }
+            parseResult = parseResult.subcommand();
+        }
+
+        return false;
+    }
+
+    protected String[] adjustLegacyArgs(String[] args) {
+        List<String> returnArgs = new ArrayList<>();
+
+
+        final ListIterator<String> iterator = Arrays.asList(args).listIterator();
+        while (iterator.hasNext()) {
+            String arg = iterator.next();
+            String argAsKey = arg.replace("-", "").toLowerCase();
+
+            if (arg.startsWith("-")) {
+                returnArgs.add(arg);
+            } else {
+                final String legacyTag = this.legacyPositionalArguments.get(argAsKey);
+                if (legacyTag == null) {
+                    returnArgs.add(arg);
+                } else {
+                    returnArgs.add(arg);
+
+                    String value = " ";
+                    while (iterator.hasNext()) {
+                        arg = iterator.next();
+                        if (arg.startsWith("-")) {
+                            iterator.previous();
+                            break;
+                        } else {
+                            value += arg + " ";
+                        }
+                    }
+
+                    value = StringUtil.trimToNull(value);
+                    if (value != null) {
+                        returnArgs.add("--" + legacyTag);
+                        returnArgs.add(value);
+                    }
+                }
+            }
+        }
+
+        return returnArgs.toArray(new String[0]);
+    }
+
+    static String[] getCommandNames(CommandLine.ParseResult parseResult) {
+        List<String> returnList = new ArrayList<>();
+        for (CommandLine command : parseResult.asCommandLineList()) {
+            final String commandName = command.getCommandName();
+            if (commandName.equals("liquibase")) {
+                continue;
+            }
+            returnList.add(commandName);
+        }
+        return returnList.toArray(new String[0]);
+    }
+
+    private List<ConfigurationValueProvider> registerValueProviders(String[] args) {
+        final LiquibaseConfiguration liquibaseConfiguration = Scope.getCurrentScope().getSingleton(LiquibaseConfiguration.class);
+        List<ConfigurationValueProvider> returnList = new ArrayList<>();
+
+        final CommandLineArgumentValueProvider argumentProvider = new CommandLineArgumentValueProvider(commandLine.parseArgs(args), legacyNoLongerCommandArguments);
+        liquibaseConfiguration.registerProvider(argumentProvider);
+        returnList.add(argumentProvider);
+
+        final File defaultsFile = new File(IntegrationConfiguration.DEFAULTS_FILE.getCurrentValue());
+        if (defaultsFile.exists()) {
```

Comment:
Can defaultsFile be null?

Reply:
defaultsFile has a default value of "liquibase.properties"